build.yaml

```diff
@@ -221,7 +224,8 @@ common_cli:
     export FLOCKER_FUNCTIONAL_TEST_OPENSTACK_REGION=dfw
 
   setup_coverage: &setup_coverage |
-    parse_logs pip install coverage==3.7.1 http://data.hybridcluster.net/python/coverage_reporter-0.01_hl0-py27-none-any.whl  --trusted-host \$TRUSTED_HOST
+    parse_logs pip install coverage  \
+      -i \$PIP_INDEX_URL --trusted-host \$TRUSTED_HOST
 
```

Comment:
\>pip2 search coverage_reporter
shows nothing, so I suppose that package is only available from data.hybridcluster.com, right ?

coverage is used by the jenkins cobertura plugin (will document this on the build.yaml), to produce coverage reports.
It can provide 'trends' that graph the changes in total line coverage over a period of time (builds), however those graphs are not really valuable on feature branches as their timespan is too short.
The trends however from the master branch may be usefull as it should contain the build history over time.
http://52.88.192.98:8080/job/ClusterHQ-flocker/job/flocker-ZFS-builds-FLOC-2853/job/run_trial_on_AWS_Ubuntu_Trusty_flocker.restapi/5/cobertura/
http://azul-jenkins:8080/job/ClusterHQ-flocker/job/flocker-ZFS-builds-FLOC-2853/job/run_trial_on_AWS_Ubuntu_Trusty_flocker.restapi/5/cobertura/

AFAIK in Jenkins I don't think there is a way to track down changes down to the PR level as you mention ( at least not my knowledge ), could look into it.
Not sure if anyone looks into those metrics today